messaging/kafka/plugin_impl_kafka.go

```diff
@@ -50,154 +52,170 @@ type Deps struct {
 
-	// Get config data
-	config := &mux.Config{}
-	found, err := p.PluginConfig.GetValue(config)
+	// Get muxCfg data (contains kafka brokers ip addresses)
+	muxCfg := &mux.Config{}
+	found, err := plugin.PluginConfig.GetValue(muxCfg)
 	if !found {
-		p.Log.Info("kafka config not found ", p.PluginConfig.GetConfigName(), " - skip loading this plugin")
-		p.disabled = true
+		plugin.Log.Info("kafka muxCfg not found ", plugin.PluginConfig.GetConfigName(), " - skip loading this plugin")
+		plugin.disabled = true
 		return nil //skip loading the plugin
 	}
 	if err != nil {
 		return err
 	}
-	clientConfig := p.getClientConfig(config, p.Log, topic)
+	// retrieve clientCfg
+	clientCfg := plugin.getClientConfig(muxCfg, plugin.Log, topic)
 
-	// Init consumer
-	p.consumer, err = client.NewConsumer(clientConfig, nil)
+	// init 'hash' sarama client
+	plugin.hsClient, err = client.NewClient(clientCfg, client.Hash)
 	if err != nil {
 		return err
 	}
 
-	// Initialize both multiplexers to allow both, dynamic and manual mode
-	if p.muxHash == nil {
-		p.muxHash, err = mux.InitMultiplexerWithConfig(config, p.ServiceLabel.GetAgentLabel(), client.Hash, p.Log)
-		if err != nil {
-			return err
-		}
-		p.Log.Debug("Default multiplexer initialized")
+	// init 'manual' sarama client
+	plugin.manClient, err = client.NewClient(clientCfg, client.Manual)
+	if err != nil {
+		return err
+	}
+
+	// init bsm/sarama consumer
+	plugin.consumer, err = client.NewConsumer(clientCfg, nil)
+	if err != nil {
+		return err
 	}
-	if p.muxManual == nil {
-		p.muxManual, err = mux.InitMultiplexerWithConfig(config, p.ServiceLabel.GetAgentLabel(), client.Manual, p.Log)
+
+	// Initialize both multiplexers to allow both, dynamic and manual mode
+	if plugin.mux == nil {
+		name := plugin.ServiceLabel.GetAgentLabel() + "-hash"
+		plugin.mux, err = mux.InitMultiplexerWithConfig(clientCfg, plugin.hsClient, plugin.manClient, name, plugin.Log)
 		if err != nil {
 			return err
 		}
-		p.Log.Debug("Manual multiplexer initialized")
+		plugin.Log.Debug("Default multiplexer initialized")
 	}
 
 	return err
 }
 
 // AfterInit is called in the second phase of the initialization. The kafka multiplexerNewWatcher
 // is started, all consumers have to be subscribed until this phase.
-func (p *Plugin) AfterInit() error {
-	if p.muxHash == nil {
-		return nil
+func (plugin *Plugin) AfterInit() error {
+	if plugin.mux != nil {
+		err := plugin.mux.Start()
+		if err != nil {
+			return err
+		}
 	}
 
 	// Register for providing status reports (polling mode)
-	if p.StatusCheck != nil {
-		p.StatusCheck.Register(p.PluginName, func() (statuscheck.PluginState, error) {
+	if plugin.StatusCheck != nil {
+		plugin.StatusCheck.Register(plugin.PluginName, func() (statuscheck.PluginState, error) {
 			// Method 'RefreshMetadata()' returns error if kafka server is unavailable
-			err := p.consumer.Client.RefreshMetadata(topic)
+			err := plugin.consumer.Client.RefreshMetadata(topic)
```

Comment:
call directly plugin.hsClient.RefreshMetadata. the consumer is unnecessary here.